tl-services/src/main/java/org/egov/tl/service/EnrichmentService.java

```diff
@@ -632,39 +630,52 @@ public void enrichAssignes(TradeLicense license){
     private void enrichCTLTradeUnits(TradeLicense license) {
     	TradeLicenseDetail licenseDetail = license.getTradeLicenseDetail();
     	String businessService = license.getBusinessService();
+    	JsonNode addtionalDetails = licenseDetail.getAdditionalDetail();
+    	Long dob = licenseDetail.getOwners().get(0).getDob();
     	TradeUnit e = null;
-    	String licensePeriod =null;
-    	switch(businessService) {
+    	int licensePeriod =0;
+    	int age=0;
+    	switch(businessService) {    		
     		case businessService_REHRI_DL:    
-    				Long dob = licenseDetail.getOwners().get(0).getDob();
     				//Getting license period using DOB.
-    				 licensePeriod = getLicensePeriod(dob);
+    				age = getAgeForLicensePeriod(dob);
+    				if(age <=50)
+    					 licensePeriod = 15;			 
+    					else
+    					 licensePeriod = 5;
     				//Updating licensePeriod in additional details.
-    				JsonNode addtionalDetails = licenseDetail.getAdditionalDetail();
-    				((ObjectNode)addtionalDetails).put("licensePeriod", licensePeriod+" years");
+    				
+    				((ObjectNode)addtionalDetails).put("licensePeriod", licensePeriod);
         			e = TradeUnit.builder()
 		    				.tradeType(businessService)
 		    				.uom("year")
-		    				.uomValue(licensePeriod)
+		    				.uomValue(String.valueOf(licensePeriod))
 		    				.build();    			
         			break;
     		case businessService_REHRI_RC:
-    			 licensePeriod = licenseDetail.getAdditionalDetail().get("licensePeriod").asText().split("years")[0];
+    			 int lPeriod = licenseDetail.getAdditionalDetail().get("licensePeriod").asInt();
+    			 age = getAgeForLicensePeriod(dob);
+    			 if(age <=50)
+					 licensePeriod = lPeriod;			 
+					else
+					 licensePeriod = 5;
     			 e = TradeUnit.builder()
```

Comment:
Don't we need to set the licensePeriod that is calculated based on age back to the license.
The following statement is missing for the case `businessService_REHRI_RC`
```
((ObjectNode)addtionalDetails).put("licensePeriod", licensePeriod);
```